File: OPENAPI-REST-API/openapi-client/javascript-closure-angular/API/Client/MailOrderRequest.js
goog.provide('API.Client.MailOrderRequest');

/**
 * Request body to validate (putMail) or place (addMail) a mail order. Service type ids come from getNewMail.serviceTypes.
 * @record
 */
API.Client.MailOrderRequest = function() {}

/**
 * Mail plan service type id from getNewMail.serviceTypes (must be buyable).
 * @type {!number}
 * @export
 */
API.Client.MailOrderRequest.prototype.serviceType;

/**
 * Coupon code.
 * @type {!string}
 * @export
 */
API.Client.MailOrderRequest.prototype.coupon;

/**
 * Free-form note saved on the service row (used on addMail).
 * @type {!string}
 * @export
 */
API.Client.MailOrderRequest.prototype.comment;

File: OPENAPI-REST-API/swagger-client/scala-akka-http-server/src/main/scala/io/swagger/server/model/MailOrderRequest.scala
package io.swagger.server.model


/**
 * Request body to validate (putMail) or place (addMail) a mail order. Service type ids come from getNewMail.serviceTypes.
 *
 * @param serviceType Mail plan service type id from getNewMail.serviceTypes (must be buyable).
 * @param coupon Coupon code.
 * @param comment Free-form note saved on the service row (used on addMail).
 */
case class MailOrderRequest (
  serviceType: Int,
  coupon: Option[String],
  comment: Option[String]
)

